controllers/controller.go

```diff
@@ -55,30 +49,7 @@ const (
 
 // FromConfiguration is a factory function to create instances of the Controller based on the service provider
 // configuration.
-func FromConfiguration(fullConfig config.Configuration, spConfig config.ServiceProviderConfiguration, kubeConfig *rest.Config) (Controller, error) {
-	authtor, err := authentication.NewFromConfig(fullConfig, kubeConfig)
-	if err != nil {
-		return nil, err
-	}
-
-	scheme := runtime.NewScheme()
-	if err = corev1.AddToScheme(scheme); err != nil {
-		return nil, err
-	}
-
-	if err = v1beta1.AddToScheme(scheme); err != nil {
-		return nil, err
-	}
-
-	cl, err := client.New(kubeConfig, client.Options{Scheme: scheme})
-	if err != nil {
-		return nil, err
-	}
-
-	vaultStorage, err := tokenstorage.NewVaultStorage("spi-oauth", fullConfig.VaultHost, fullConfig.ServiceAccountTokenFilePath)
-	if err != nil {
-		return nil, err
-	}
+func FromConfiguration(fullConfig config.Configuration, spConfig config.ServiceProviderConfiguration, sessionManager *scs.Manager, cl AuthenticatingClient, vaultStorage tokenstorage.TokenStorage, redirectTemplate *template.Template) (Controller, error) {
```

Comment:
please rename `vaultStorage` here. It should not depend on implementation, right? It can be any storage.

Reply:
👍🏼